### PR TITLE
feat(tiles): detect content formats

### PR DESCRIPTION
**Pull request by belom88**

### CODE REVIEW

File: modules/gltf/src/lib/api/gltf-scenegraph.ts [hunk 1]
```diff
@@ -21,11 +21,19 @@ import {
   getAccessorTypeFromSize,
   getComponentTypeFromArray
 } from '../gltf-utils/gltf-utils';
+import {name as dracoExtensionName} from '../extensions/KHR_draco_mesh_compression';
```
Comment (ibgreen) [inline modules/gltf/src/lib/api/gltf-scenegraph.ts:24]: Maybe use ALL_CAPS for constants?

File: modules/gltf/src/lib/api/gltf-scenegraph.ts [hunk 2]
```diff
@@ -21,11 +21,19 @@ import {
...
+import {name as meshoptExtensionName} from '../extensions/EXT_meshopt_compression';
+import {name as ktx2ExtensionName} from '../extensions/KHR_texture_basisu';
 
 type GLTFWithBuffers = {
   json: GLTF;
   buffers: any[];
   binary?: ArrayBuffer;
+  contentFormats?: {
```
Comment (ibgreen) [inline modules/gltf/src/lib/api/gltf-scenegraph.ts:32]: Is it worth nesting these?

**Review by ibgreen — COMMENTED** [review]
Just for discussion. 

`GLTFWithBuffers` is more of an input data class, and this information is discovered during the parsing and postprocessing, so it seems odd to have these here. The processing returns a GLTFScenegraph, so that seems like a more appropriate place).

But on GLTFScenegraph we already have a convenient API to check for these extensions. 

Is the problem that it is inconvenient to import those extension names? We could add them as static members on the GLTFScenegraph instead?
```
class GLTFScenegraph {
  static DRACO_EXTENSION: string = DRACO_EXTENSION_NAME;
}
```
Then we can just do 
```
scenegraph.getExtension(GLTFScenegraph.DRACO_EXTENSION);
```

Do we need to extract this into an additional flag?

Maybe just add additional methods to the scenegraph class?

```
usesDraco()
usesMeshopt()
usesKTX2()
```

**Comment by belom88** [comment]
The problem of gaining thouse formats is in extensions. They just create the instance of `GLTFScenegraph` and use it as wrapper for modifying `GLTFWithBuffers` inside. As the result the extensions mutate this input `GLTFWithBuffers`. All other data created inside the instance of `GLTFScenegraph` is forgottent in the scope of `decode` function. During decode every extension removes itself from the json. Eg.:
https://github.com/visgl/loaders.gl/blob/29b08f3519c50984e84bf4234e607cab7c7d1c3e/modules/gltf/src/lib/extensions/KHR_draco_mesh_compression.ts#L60
So after an extension `decode` we are not able to get the input content format.

In this logic the only way to save information about used extension and pass it to `post-process` is mutating `GLTFWithBuffers`

**Comment by ibgreen** [comment]
I see. The idea is to decode and then remove the extensions flags so that the next stage doesn't need to even know that those formats exist.

But I see that the problem you are trying to solve is that you want to keep information about whether those were present in the input. 

Since they are removed in the `GLTFScenegraph` class I think it could make sense to track them there. We could move your flags object into that class. Perhaps a more general approach would be to just add a `removedExtensions` or `processedExtensions` to the root object.

**Review by ibgreen — APPROVED** [review]
I pushed a commit that simplifies things a little bit. We just add the tracking to removeExtension. Hopefully this still works for you. Feel free to land.

**Pull request merged**